tests/unit/s2n_safety_blinding_test.c

```diff
@@ -0,0 +1,180 @@
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, RESULT_GUARD_PTR(s2n_ptr_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, RESULT_GUARD(s2n_result_func(false)));
+    return S2N_RESULT_OK;
+}
+
+int s2n_posix_test(struct s2n_connection *conn)
+{
+    WITH_ERROR_BLINDING(conn, POSIX_GUARD_RESULT(s2n_result_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, POSIX_ENSURE(true, S2N_ERR_UNIMPLEMENTED));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, POSIX_GUARD(s2n_posix_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, POSIX_GUARD_PTR(s2n_ptr_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, POSIX_GUARD(s2n_posix_func(false)));
+    return S2N_SUCCESS;
+}
+
+int* s2n_ptr_test(struct s2n_connection *conn)
+{
+    WITH_ERROR_BLINDING(conn, PTR_GUARD_RESULT(s2n_result_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, PTR_ENSURE(true, S2N_ERR_UNIMPLEMENTED));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, PTR_GUARD_POSIX(s2n_posix_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, PTR_GUARD(s2n_ptr_func(true)));
+    EXPECT_NO_BLINDING(conn);
+
+    WITH_ERROR_BLINDING(conn, PTR_GUARD(s2n_ptr_func(false)));
+    return S2N_SUCCESS;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test: s2n_connection_apply_error_blinding */
+    {
+        /* Safety check */
+        struct s2n_connection *conn = NULL;
+        EXPECT_ERROR_WITH_ERRNO(s2n_connection_apply_error_blinding(NULL), S2N_ERR_NULL);
+        EXPECT_OK(s2n_connection_apply_error_blinding(&conn));
+
+        conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(conn);
+
+        /* No-op for no error */
+        {
+            SETUP_TEST(conn);
+            s2n_errno = S2N_ERR_OK;
+            EXPECT_OK(s2n_connection_apply_error_blinding(&conn));
+            EXPECT_NO_BLINDING(conn);
+        }
+
+        /* No-op for retriable errors */
+        {
+            SETUP_TEST(conn);
+            s2n_errno = S2N_ERR_IO_BLOCKED;
+            EXPECT_OK(s2n_connection_apply_error_blinding(&conn));
+            EXPECT_NO_BLINDING(conn);
+        }
+
+        /* Closes connection but does not blind for non-blinding errors */
+        {
+            SETUP_TEST(conn);
+            s2n_errno = S2N_ERR_CIPHER_NOT_SUPPORTED;
+            EXPECT_OK(s2n_connection_apply_error_blinding(&conn));
+            EXPECT_EQUAL(s2n_connection_get_delay(conn), 0);
+            EXPECT_TRUE(conn->closed);
+        }
+
+        /* Blinds for an average error */
+        {
+            SETUP_TEST(conn);
+            s2n_errno = S2N_ERR_UNIMPLEMENTED;
+            EXPECT_OK(s2n_connection_apply_error_blinding(&conn));
+            EXPECT_BLINDING(conn);
+        }
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test: WITH_ERROR_BLINDING macro */
```

Comment:
I could use more of a descriptive comment about what this is testing. It's hard to follow, given WITH_ERROR_BLINDING isn't directly being called in this test.